examples/esplora_backend.rs

```diff
@@ -0,0 +1,93 @@
+use std::str::FromStr;
+
+use bdk::blockchain::Blockchain;
+use bdk::{
+    blockchain::esplora::EsploraBlockchain,
+    database::MemoryDatabase,
+    template::Bip84,
+    wallet::{export::FullyNodedExport, AddressIndex},
+    KeychainKind, SyncOptions, Wallet,
+};
+use bitcoin::{
+    util::bip32::{self, ExtendedPrivKey},
+    Network,
+};
+
+pub mod utils;
+
+use crate::utils::tx::build_signed_tx;
+
+/// This will create a wallet from an xpriv and get the balance by connecting to an Esplora server.
+/// If enough amount is available, this will send a transaction to an address.
+/// Otherwise, this will display a wallet address to receive funds.
+///
+/// This can be run with `cargo run --features="use-esplora-reqwest, reqwest-default-tls" --example esplora_backend`
+/// in the root folder.
+///
+/// Note: The configuration above uses asynchronous HTTP calls.
```

Comment:
Sorry I have missed this before. Nothing major, feel free to ignore for this PR. 

Even though the calls are async in `reqwest` they are still blocking in our impl, using the `await_or_block` macro like this, https://github.com/bitcoindevkit/bdk/blob/06310f1dd06641c36cb1980ee34489e03eadf96b/src/blockchain/esplora/reqwest.rs#L126 

So to actually make the wallet calls async in the example you need to use the `async-interface` feature too.. In that case you will have `await` in your wallets calls in the example.. 

This PR is good to go on its own.. Maybe as a future PR an idea.

 - Transform this to a blocking example using `ureq`.
 - Create another fully `async` example using `reqwest`.